### ISSUE TITLE
Access to multithreading?

### ISSUE THREAD

**Issue by ioannisPApapadopoulos** [JuliaApproximation/FastTransforms.jl#250]
On my machine multithreading seems to _slow down_ the code and I assume I am missing a trick.

MWE: 

```
using FastTransforms
c = rand(10_000,16)
F = plan_cheb2leg(c)
F*c;
@time F*c;
```

```
julia -t 1 > 0.015957 seconds (2 allocations: 1.221 MiB)
julia -t 2 > 0.017736 seconds (2 allocations: 1.221 MiB)
julia -t 4 > 0.021904 seconds (2 allocations: 1.221 MiB)
julia -t 16 > 0.014899 seconds (2 allocations: 1.221 MiB)

```
@MikaelSlevinsky do you have any advice? Thank you!

**Comment by TSGut** [comment]
Isn't this within variance? For fast things like this BenchmarkTools with @btime is usually better since it runs it a lot of times.

**Comment by ioannisPApapadopoulos** [comment]
Similar issue:

```
using FastTransforms
using BenchmarkTools
c = rand(10_000,16)
F = plan_cheb2leg(c)
@btime F*c;
```

```
julia -t 1 > 14.509 ms (2 allocations: 1.22 MiB)
julia -t 2 > 14.733 ms (2 allocations: 1.22 MiB)
julia -t 4 > 14.372 ms (2 allocations: 1.22 MiB)
julia -t 16 > 14.587 ms (2 allocations: 1.22 MiB)
```

**Comment by jishnub** [comment]
I'm unsure if this package uses threading by itself. Perhaps you may look into threading in FFTW

**Comment by ioannisPApapadopoulos** [comment]
Oh really? @dlfivefifty gave me a different impression in https://github.com/JuliaApproximation/ClassicalOrthogonalPolynomials.jl/issues/204

**Comment by MikaelSlevinsky** [comment]
FastTransforms uses multithreading via OpenMP. It uses a similar model to FFTW.jl, where the threading occurs inside the C library, not through Julia. Therefore, you don't need to start Julia with a certain number of threads. This code sets the maximum number of threads upon initialization to a typical number of cores ([hyperthreading is irrelevant for compute-bound computations](https://discourse.julialang.org/t/hyperthreading-in-hpc/101316)) and subsequently allows the Julia user to control the number of threads:
https://github.com/JuliaApproximation/FastTransforms.jl/blob/d8cceddd9e9b0829602c2a830de2588655754ea6/src/libfasttransforms.jl#L12-L20
Evidence with 18 cores:
```julia
julia> using FastTransforms

julia> c = randn(10_000, 16);

julia> p = plan_cheb2leg(c);

julia> for t in (1, 2, 4, 8, 16)
           FastTransforms.ft_set_num_threads(t)
           @time p*c
       end
  0.123884 seconds (2 allocations: 1.221 MiB)
  0.070071 seconds (2 allocations: 1.221 MiB)
  0.032233 seconds (2 allocations: 1.221 MiB)
  0.017220 seconds (2 allocations: 1.221 MiB)
  0.010281 seconds (2 allocations: 1.221 MiB)

```
So, in summary, threading is already built in.

**Comment by dlfivefifty** [comment]
> Oh really? @dlfivefifty gave me a different impression

No I didn't! Perhaps you meant to say "@dlfivefifty told me that FastTransforms has multithreading built in but I chose to jump to the conclusion that it uses Julia's thread pool" 😛

**Comment by ioannisPApapadopoulos** [comment]
Thanks @MikaelSlevinsky for the explanation!
